File: HospitalManagementSystem/Hospital.Domain/Models/Doctor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Hospital.Domain.Models
{
    public class Doctor
    {
        [Column("DoctorId")]
        public int Id { get; set; }

        [MinLength(1)]
        [MaxLength(50)]
        public required string Name { get; set; }

        [MinLength(1)]
        [MaxLength(50)]
        public required string Surname { get; set; }

        [MinLength(5)]
        [MaxLength(100)]
        public required string Address { get; set; }

        [MinLength(5)]
        [MaxLength(20)]
        public required string PhoneNumber { get; set; }

        public required Department Department { get; set; }
        public ICollection<DoctorsPatients> DoctorsPatients { get; set; } = [];

        public required DoctorSchedule WorkingHours { get; set; }
    }
}

File: HospitalManagementSystem/Hospital.App/src/models/Doctor.ts
import { PatientPartial } from "./PatientPartial";

export interface Doctor {
  id: number;
  name: string;
  surname: string;
  address: string;
  phoneNumber: string;
  department: string;
  workingHours?: DoctorSchedule;
  patients?: PatientPartial[];
}

interface DoctorSchedule {
  startShift: string;
  endShift: string;
  weekDays: string[];
}
